src/extractor/guidance/segregated_intersection_classification.cpp

```diff
@@ -145,18 +55,124 @@ std::unordered_set<EdgeID> findSegregatedNodes(const NodeBasedGraphFactory &fact
         return length;
     };
 
-    auto const get_edge_info = [&](NodeID node, auto const &edgeData) -> EdgeInfo {
+    // Returns an angle between edges from from_edge_id to to_edge_id
+    auto const get_angle = [&](NodeID from_node, EdgeID from_edge_id, NodeID to_edge_id) {
+        auto intersection_node = graph.GetTarget(from_edge_id);
+        auto from_edge_id_outgoing = graph.FindEdge(intersection_node, from_node);
+        auto to_node = graph.GetTarget(to_edge_id);
+        auto const node_to =
+            coordExtractor.GetCoordinateCloseToTurn(intersection_node, to_edge_id, false, to_node);
+        auto const node_from = coordExtractor.GetCoordinateCloseToTurn(
+            intersection_node, from_edge_id_outgoing, false, from_node);
+        return util::coordinate_calculation::computeAngle(
+            node_from, coordinates[intersection_node], node_to);
+    };
+
+    auto const get_edge_info = [&](EdgeID edge_id, NodeID node, auto const &edgeData) -> EdgeInfo {
         /// @todo Make string normalization/lowercase/trim for comparison ...
 
         auto const id = annotation[edgeData.annotation_data].name_id;
         BOOST_ASSERT(id != INVALID_NAMEID);
         auto const name = names.GetNameForID(id);
 
-        return {node,
+        return {edge_id,
+                node,
                 name,
+                edgeData.reversed,
                 edgeData.reversed ? 1 : 0,
                 annotation[edgeData.annotation_data].classes,
-                edgeData.flags.road_classification.GetClass()};
+                edgeData.flags};
+    };
+
+    auto is_bidirectional = [](auto flags) { return flags.is_split || (!flags.is_split && flags.forward && flags.backward); };
+
+    auto isSegregated = [&](NodeID node1,
+                            std::vector<EdgeInfo> v1,
+                            std::vector<EdgeInfo> v2,
+                            EdgeInfo const &current,
+                            double edgeLength) {
+        // Internal intersection edges must be short and cannot be a roundabout
+        // TODO adjust length as needed with lamda
+        if (edgeLength > 32.0f || current.flags.roundabout) {
+            return false;
+        }
+
+        // Print information about angles at node 1 from all inbound edge to the current edge
+        bool oneway_inbound = false;
+        for (auto const &edge_from : v1) {
```

Comment:
Reading the two four loops, it seems like both for loops check the same things on different data. Maybe we could introduce a helper function that we could reuse here?